File: models/Vehiculo.cs
namespace TallerMecanico.models;

public class Vehiculo
{
    public int Id { get; set; }
    public int ClienteId { get; set; }
    public Cliente Cliente { get; set; }
    public string Marca { get; set; }
    public string Modelo { get; set; }
    public int Año { get; set; }
    public string Placa { get; set; }

    // Relación con los servicios realizados al vehículo
    public ICollection<Servicio> ServiciosRealizados { get; set; }

    // Auditoría
    public DateTime FechaCreacion { get; set; }
    public DateTime? FechaActualizacion { get; set; }
    public DateTime? FechaBorrado { get; set; }
    public bool EstaBorrado { get; set; } = false;
}

File: AFront/src/app/Entity/Vehiculo.ts
import { Cliente } from './Cliente';
import { Servicio } from './Servicio';

export interface Vehiculo {
  id: number;
  clienteId: number;
  cliente: Cliente | null;
  marca: string;
  modelo: string;
  //posible problema ojo
  year: number;
  placa: string;
  serviciosRealizados: Servicio[];
  fechaCreacion: Date;
  fechaActualizacion: Date | null;
  fechaBorrado: Date | null;
  estaBorrado: boolean;
}
